xla/stream_executor/gpu/gpu_command_buffer.cc

```diff
@@ -234,9 +234,37 @@ tsl::Status GpuCommandBuffer::MemcpyDeviceToDevice(DeviceMemoryBase* dst,
   if (state_ == State::kCreate) {
     Dependencies deps = GetDependencies();
     GpuGraphNodeHandle* node = &nodes_.emplace_back();
-    return GpuDriver::GraphAddMemcpyD2DNode(
-        parent_->gpu_context(), node, graph_, absl::MakeSpan(deps),
-        AsDevicePtr(*dst), AsDevicePtr(src), size);
+    return GpuDriver::GraphAddMemcpyD2DNode(parent_->gpu_context(), node,
+                                            graph_, deps, AsDevicePtr(*dst),
+                                            AsDevicePtr(src), size);
+
+    return UnsupportedStateError(state_);
+  }
+}
+
+tsl::Status GpuCommandBuffer::MemallocOnDevice(uint64_t size, DeviceMemoryBase* dptr) {
+  TF_RETURN_IF_ERROR(CheckNotFinalized());
+
+  // Adds a new memalloc node to the graph under construction.
+  if (state_ == State::kCreate) {
+    Dependencies deps = GetDependencies();
+    GpuGraphNodeHandle* node = &nodes_.emplace_back();
+    GpuDevicePtr gpu_dptr;
+    TF_RETURN_IF_ERROR(GpuDriver::GraphAddMemAllocNode(
+        node, graph_, deps, GpuDriver::MemAccessFlags::kReadWrite,
+        GpuDriver::MemLocationType::kDevice, parent_->device_ordinal(),
+        GpuDriver::MemAllocationType::kPinned, size, &gpu_dptr));
+    // For CUDA impl, VA range is reserved when adding memory allocation node.
+    CHECK(gpu_dptr);
```

Comment:
checks are discouraged in new code, if you want to keep it as an error, than add an error message explaining why it's so important (I suspect nullptr means a bug in CUDA?)

Reply:
Add error message for CHECK